x-pack/plugins/maps/public/components/layer_addpanel/view.js

```diff
@@ -4,166 +4,169 @@
+    }, () => this.props.addImportedLayer(this.state.layer));
   };
 
-  _clearSource = () => {
+  _clearLayerData = ({ keepSourceType = false }) => {
     this.setState({
       layer: null,
-      sourceType: null
+      ...(
+        !keepSourceType
+          ? { sourceType: null, importView: false }
+          : {}
+      ),
     });
     this.props.removeTransientLayer();
   }
 
-  _onSourceTypeChange = (sourceType) => {
-    this.setState({ sourceType });
+  _onSourceSelectionChange = ({ type, indexReadyFile }) => {
+    this.setState({ sourceType: type, importView: indexReadyFile });
   }
 
-  _renderNextBtn() {
-    if (!this.state.sourceType) {
-      return null;
+  _layerAddHandler = () => {
+    const layerSource = this.state.layer.getSource();
+    const boolIndexLayer = layerSource.shouldBeIndexed();
+    this.setState({ layer: null });
+    if (boolIndexLayer && !this.state.indexingTriggered) {
+      this.setState({ indexingTriggered: true });
+    } else {
+      this.props.selectLayerAndAdd();
     }
-
-    const {  hasLayerSelected, isLoading, selectLayerAndAdd } = this.props;
-    return (
-      <EuiButton
-        disabled={!hasLayerSelected}
-        isLoading={hasLayerSelected && isLoading}
-        iconSide="right"
-        iconType={'sortRight'}
-        onClick={() => {
-          this.setState({ layer: null });
-          selectLayerAndAdd();
-        }}
-        fill
-      >
-        <FormattedMessage
-          id="xpack.maps.addLayerPanel.addLayerButtonLabel"
-          defaultMessage="Add layer"
-        />
-      </EuiButton>
-    );
   }
 
-  _renderSourceCards() {
-    return ALL_SOURCES.map(Source => {
-      const icon = Source.icon
-        ? <EuiIcon type={Source.icon} size="l" />
-        : null;
+  _renderAddLayerPanel() {
+    if (!this.state.sourceType) {
       return (
-        <Fragment key={Source.type}>
-          <EuiSpacer size="s" />
-          <EuiCard
-            className="mapLayerAddpanel__card"
-            title={Source.title}
-            icon={icon}
-            onClick={() => this._onSourceTypeChange(Source.type)}
-            description={Source.description}
-            layout="horizontal"
-            data-test-subj={_.camelCase(Source.title)}
-          />
-        </Fragment>
+        <SourceSelect updateSourceSelection={this._onSourceSelectionChange} />
       );
-    });
-  }
-
-  _renderSourceSelect() {
-    return (
-      <Fragment>
-        <EuiTitle size="xs">
-          <h2>
-            <FormattedMessage
-              id="xpack.maps.addLayerPanel.chooseDataSourceTitle"
-              defaultMessage="Choose data source"
-            />
-          </h2>
-        </EuiTitle>
-        {this._renderSourceCards()}
-      </Fragment>
-    );
+    } else if (this.state.importView) {
+      return (
+        <ImportEditor
+          clearSource={this._clearLayerData}
+          previewLayer={source => this._viewLayer(source, this.state.importView)}
```

Comment:
This is called from within a check for `this.state.importView` so you can just write   `previewLayer={source => this._viewLayer(source, true)}` to be clearer that all ImportEditor's fit to bounds